### PR TITLE
Add Python interface for IndexReaderUtils methods

### PR DESCRIPTION
**Pull request by zeynepakkalyoncu**

### CODE REVIEW

File: pyserini/index/pyutils.py [hunk 1]
```diff
@@ -35,13 +35,105 @@ class IndexUtils:
...
+            if self.positions:
+                repr += ' [' + ','.join([str(p) for p in self.positions]) + ']'
+            return repr
+
+    def analyze_term(self, term):
+        '''
+        Parameters
+        ----------
+        term : str
+            Search term
```
Comment (lintool) [inline pyserini/index/pyutils.py]: just "term" - not necessary search, I think? Same below.

File: pyserini/index/pyutils.py [hunk 2]
```diff
@@ -35,13 +35,105 @@ class IndexUtils:
...
+            Stemmed term
+        '''
+        return self.object.analyzeTerm(self.reader, JString(term))
+
+    def get_term_counts(self, term):
+        '''
+        Parameters
+        ----------
+        term : str
+            Search term
```
Comment (lintool) [inline pyserini/index/pyutils.py]: note that this is before analysis/stemming?

File: pyserini/index/pyutils.py [hunk 3]
```diff
@@ -35,13 +35,105 @@ class IndexUtils:
...
+        for posting in postings_list.toArray():
+            result.append(self.Posting(posting.getDocid(), posting.getTF(), posting.getPositions()))
+        return result
+
+    def get_document_vector(self, docid):
+        '''
+        Parameters
+        ----------
+        docid : str
+            Lucene document ID
```
Comment (lintool) [inline pyserini/index/pyutils.py]: I believe this is the docid as defined in the collection? call it "collection docid" to be unambiguous. This is different from the Lucene internal docid.

So "Lucene document ID" is ambiguous.
Also below.

File: pyserini/index/pyutils.py (hunk 2 above)
Comment (zeynepakkalyoncu) [reply pyserini/index/pyutils.py]: True, I just followed the order in Anserini. Should I change both then?

Comment (lintool) [reply pyserini/index/pyutils.py]: sorry, wasn't clear - I meant, "term prior to stemming/analysis"

Comment (zeynepakkalyoncu) [reply pyserini/index/pyutils.py]: Ahh sorry, my bad... Fixed.

**Pull request merged**